src/testhost.x86/DefaultEngineInvoker.cs

```diff
@@ -83,6 +90,17 @@ public void Invoke(IDictionary<string, string> argsDictionary)
     {
         InitializeEqtTrace(argsDictionary);
 
+        // We don't have a way to pass these values to TestRequestHandler directly
+        // beacuse of it's public interface, we work around that by making it implement a second interface
+        if (_requestHandler is IDeploymentAwareTestRequestHandler deployedHandler)
+        {
+            if (argsDictionary.ContainsKey(RemotePath) && argsDictionary.ContainsKey(LocalPath))
+            {
+                deployedHandler.LocalPath = argsDictionary[LocalPath];
+                deployedHandler.RemotePath = argsDictionary[RemotePath];
+            }
+        }
```

Comment:
Inner if could be merged with outer.

```suggestion
        if (_requestHandler is IDeploymentAwareTestRequestHandler deployedHandler && 
            argsDictionary.ContainsKey(RemotePath) && 
            argsDictionary.ContainsKey(LocalPath))
        {
            deployedHandler.LocalPath = argsDictionary[LocalPath];
            deployedHandler.RemotePath = argsDictionary[RemotePath];
        }
```

Reply:
It could, but for me the current one just reads better, you know that if the type is not `IDeploymentAwareTestRequestHandler` you can stop looking at it immediately. 

Side note, do we have rule for placement of &&, to me it reads much easier if it is at the start of next line.

Reply:
> It could, but for me the current one just reads better,

Fine by me.



> Side note, do we have rule for placement of &&, to me it reads much easier if it is at the start of next line.

We don't. I need to look if there is something in StyleCop for that. I am really happy to see you also prefer them at the beginning :)